pkg/wwan/usr/bin/wwan-init.sh

```diff
@@ -439,7 +439,7 @@ collect_network_status() {
     # (e.g. radio-silence mode is switched ON/OFF) so that the updated status is promptly
     # published for better user experience.
     PROVIDERS="$("${PROTOCOL}_get_providers")"
-  else
+  elif [ "$QUERY_VISIBLE_PROVIDERS" = "true" ]; then
```

Comment:
This will affect only the case when the event is not `LONG-PROBE`. If the event is LONG-PROBE it will fetch the providers, regardless of the value of the `QUERY_VISIBLE_PROVIDERS` var. Is it correct?
I would expect something like
```bash
 if [ "$EVENT" = "LONG-PROBE" && "$QUERY_VISIBLE_PROVIDERS" = "true"]; then
```
earlier in the code. But I cannot say I totally understand what's happening =D

Reply:
Or... If we do not set the QUERY_VISIBLE_PROVIDERS, we skip even the "preserve" part during the "not-long" events, but during the long event we still get the actual list?

Reply:
There are two parts to this:
- [here](https://github.com/milan-zededa/eve/blob/disable-visible-providers/pkg/wwan/usr/bin/wwan-init.sh#L704) we make sure that LONG_PROBE is never triggered when QUERY_VISIBLE_PROVIDERS is false
- the place that your comments are referencing - if querying of visible providers is enabled, we want to preserve the list of obtained providers during "quick" and "standard" probes. But if this feature is disabled, we do not want to preserve this increasingly more obsolete information. I have added comment to this place to make the code more clear.

Reply:
Note that here is some info about these probes: https://github.com/milan-zededa/eve/blob/disable-visible-providers/pkg/wwan/usr/bin/wwan-init.sh#L25-L31